applications/StructuralMechanicsApplication/custom_response_functions/adjoint_conditions/point_load_adjoint_condition.cpp

```diff
@@ -0,0 +1,287 @@
+
+namespace Kratos
+{
+    //******************************* CONSTRUCTOR ****************************************
+    //************************************************************************************
+
+    PointLoadAdjointCondition::PointLoadAdjointCondition( IndexType NewId, GeometryType::Pointer pGeometry )
+        : PointLoadCondition( NewId, pGeometry )
+    {
+        //DO NOT ADD DOFS HERE!!!
+    }
+
+    //************************************************************************************
+    //************************************************************************************
+
+    PointLoadAdjointCondition::PointLoadAdjointCondition( IndexType NewId, GeometryType::Pointer pGeometry,  PropertiesType::Pointer pProperties )
+        : PointLoadCondition( NewId, pGeometry, pProperties )
+    {
+    }
+
+    //********************************* CREATE *******************************************
+    //************************************************************************************
+
+    Condition::Pointer PointLoadAdjointCondition::Create(IndexType NewId,GeometryType::Pointer pGeom,PropertiesType::Pointer pProperties) const
+    {
+        return boost::make_shared<PointLoadAdjointCondition>(NewId, pGeom, pProperties);
+    }
+
+    //************************************************************************************
+    //************************************************************************************
+
+    Condition::Pointer PointLoadAdjointCondition::Create( IndexType NewId, NodesArrayType const& ThisNodes,  PropertiesType::Pointer pProperties ) const
+    {
+        return boost::make_shared<PointLoadAdjointCondition>( NewId, GetGeometry().Create( ThisNodes ), pProperties );
+    }
+
+    //******************************* DESTRUCTOR *****************************************
+    //************************************************************************************
+
+    PointLoadAdjointCondition::~PointLoadAdjointCondition()
+    {
+    }
+
+      //************************************************************************************
+    //************************************************************************************
+
+    void PointLoadAdjointCondition::EquationIdVector(EquationIdVectorType& rResult, ProcessInfo& rCurrentProcessInfo )
+    {
+        KRATOS_TRY
+
+        const unsigned int NumberOfNodes = GetGeometry().size();
+        const unsigned int dim = GetGeometry().WorkingSpaceDimension();
+        if (rResult.size() != dim * NumberOfNodes)
+        {
+            rResult.resize(dim*NumberOfNodes,false);
+        }
+
+        const unsigned int pos = this->GetGeometry()[0].GetDofPosition(ADJOINT_DISPLACEMENT_X);
+
+        if(dim == 2)
+        {
+            for (unsigned int i = 0; i < NumberOfNodes; ++i)
+            {
+                const unsigned int index = i * 2;
+                rResult[index    ] = GetGeometry()[i].GetDof(ADJOINT_DISPLACEMENT_X,pos    ).EquationId();
```

Comment:
if the code from PointLoad  is the same except for this maybe we can give the variable on the constructor to avoid code duplication. Just suggestion

Reply:
well, just take care NOT TO STORE anything. If you need storage then derive